common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -809,4 +811,21 @@ private boolean cachedAccessTokenKidMatchesKeystoreKid(@Nullable final String at
 
         return deviceKid.equals(atKid);
     }
+
+    /**
+     * Using this method to load the {@link OpenIdProviderConfiguration}
+     */
+    public void loadOpenIdProviderConfiguration() {
+        try {
+            final OpenIdProviderConfigurationClient client =
+                    new OpenIdProviderConfigurationClient();
+            mOpenIdProviderConfiguration = client.loadOpenIdProviderConfigurationFromAuthority(mConfig.getAuthorityUrl().toString());
+        } catch (ServiceException e) {
+            Logger.error(
```

Comment:
Do w need to throw in this case? What is the expected behavior for an Acquire Token call when this exception occurs?

Reply:
I think it's best to fallback to the default way we did it where we use the default authorization end point, as that may still work in some cases (although probably would still fail, if the authority url does not have an Open ID Configuration). I could go either way. Since it's probably going to fail anyway, maybe we can just throw a client exception. Thoughts?